src/routes/test-contentful-content/+page.server.ts

```diff
@@ -7,24 +7,60 @@ import type { Document } from "@contentful/rich-text-types";
 import type { EntryQuery } from "./$queries.generated";
 
 const query = gql`
+  fragment ImageProps on Asset {
+    sys {
+      id
+    }
+    contentType
+    title
+    description
+    url
+    width
+    height
+  }
+
   query Entry {
     testRichText(id: "V7ibT9I8Vg99iKsDgLhsK") {
       title
       body {
         json
+        links {
+          assets {
+            block {
+              ...ImageProps
+            }
+            hyperlink {
+              ...ImageProps
+            }
+          }
+        }
       }
     }
   }
 `;
 
-export const load = async () => {
+export const load = async ({ fetch }) => {
   const client = getContentfulClient({
     spaceID: CONTENTFUL_SPACE_ID,
     token: CONTENTFUL_DELIVERY_API_TOKEN,
   });
   const data = await client.fetch<EntryQuery>(printQuery(query));
+  const blurhashes = Object.fromEntries(
+    await Promise.all(
+      data?.testRichText?.body?.links.assets.block
+        .filter((item) => !!item)
+        .flatMap(async (item) => {
+          if (!item?.sys?.id || !item?.url || !item?.contentType?.startsWith("image/")) return [];
+          const blurhashResponse = await fetch(`/api/v1/blurhash/${encodeURIComponent(item.url)}`);
+          if (!blurhashResponse.ok) return [];
+          return [item.sys.id, await blurhashResponse.text()];
+        }) ?? []
+    )
+  );
```

Comment:
Is this something we'll need to do in all of our `+page.server.ts` files, or can this be integrated into the Contentful client?

Reply:
Hmm. The issue with trying to make the Contentful client do it transparently is that it would need to either walk the entire response to every GraphQL request to check for images or rich text, or we would have to tell it about the structure of the query somehow. Since we're parsing the queries at build time and using them as data instead of as strings, it _might_ be possible to do this automatically. Even if that isn't possible, this should get refactored into a helper at the very least.

What I _wish_ we could do is request it in the image component itself, and then have SvelteKit await that on the server before responding to the request. Unfortunately, async server-side rendering is not (yet?) available in SvelteKit.

I'll refactor this into some helper functions to get this merged, and look into doing it transparently in the Contentful client in future work.